src/Core/TileGeometry.ts

```diff
@@ -0,0 +1,157 @@
+import * as THREE from 'three';
+
+import { computeBuffers, getBufferIndexSize }
+    from 'Core/Prefab/computeBufferTileGeometry';
+import { GpuBufferAttributes, Projected, TileBuilder, TileBuilderParams }
+    from 'Core/Prefab/TileBuilder';
+import Extent from 'Core/Geographic/Extent';
+import Cache from 'Core/Scheduler/Cache';
+
+import OBB from 'Renderer/OBB';
+
+type PartialTileBuilderParams =
+    Pick<TileBuilderParams, 'extent' | 'level' | 'zoom'>
+    & Partial<TileBuilderParams>;
+
+function defaultBuffers(
+    builder: TileBuilder<TileBuilderParams>,
+    params: PartialTileBuilderParams,
+): GpuBufferAttributes {
+    const fullParams = {
+        disableSkirt: false,
+        hideSkirt: false,
+        buildIndexAndUv_0: true,
+        segments: 16,
+        projected: new Projected(0, 0),
+        center: builder.center(params.extent!).clone(),
+        ...params,
+    };
+
+    const buffers = computeBuffers(builder, fullParams);
+    const bufferAttributes = {
+        index: buffers.index
+            ? new THREE.BufferAttribute(buffers.index, 1)
+            : null,
+        uvs: [
+            ...(buffers.uvs[0]
+                ? [new THREE.BufferAttribute(buffers.uvs[0], 2)]
+                : []
+            ),
+            ...(buffers.uvs[1]
+                ? [new THREE.BufferAttribute(buffers.uvs[1], 1)]
+                : []),
+        ],
+        position: new THREE.BufferAttribute(buffers.position, 3),
+        normal: new THREE.BufferAttribute(buffers.normal, 3),
+    };
+
+    return bufferAttributes;
+}
+
+export class TileGeometry extends THREE.BufferGeometry {
+    public OBB: OBB | null;
+    public extent: Extent;
+    public segments: number;
+    public tileCenter: THREE.Vector3;
+
+    private _refCount: {
+        count: number,
+        fn: () => void,
+    } | null;
+
+    public constructor(
+        builder: TileBuilder<TileBuilderParams>,
```

Comment:
The design pattern is not really clear here... There are some circular dependencies and I think that separation of concerns is not really effective. I think we should aim at: (and we are not far from it)
* interface TileBuilder
* implemented by PlanarTileBuilder and GlobeTileBuilder
* each having a method returning a new TileGeometry based on relevant TileBuilderParams (the current newTileGeometry)
* TileGeometry being only a data model that does not know about the TileBuilder.

WDYT @Desplandis @HoloTheDrunk?

(Note that we don't necessarily have to implement it in this PR)

Reply:
Letting the builders have their own implementation is the most natural implementation to me as well, since it would also allow us to bind the tile cache currently stored in a global variable inside builder instances, enabling the use of multiple builders in the same program without the issues that currently poses (though I don't know if there is a use case for that possibility).

Reply:
Ok, I created #2470 so we don't forget about it